qiskit/utils/run_circuits.py

```diff
@@ -457,81 +457,126 @@ def run_circuits(
     run_config = run_config or {}
     with_autorecover = not is_simulator_backend(backend)
 
-    job, job_id = _safe_submit_circuits(
-        circuits,
-        backend,
-        qjob_config=qjob_config,
-        backend_options=backend_options,
-        noise_config=noise_config,
-        run_config=run_config,
-    )
-    result = None
+    if MAX_CIRCUITS_PER_JOB is not None:
+        max_circuits_per_job = int(MAX_CIRCUITS_PER_JOB)
+    else:
+        if is_local_backend(backend):
+            max_circuits_per_job = sys.maxsize
```

Comment:
I just copied this from the qobj path, but I don't think this is actually correct. We should probably defer to the `max_experiments` value in the configuration because a local backend might not accept the max list size circuits for a single job.